src/Components/ABDM/LinkAbhaNumber/CreateWithAadhaar.tsx

```diff
@@ -0,0 +1,618 @@
+  );
+}
+
+type IVerifyMobileNumberProps = InjectedStepProps<Memory>;
+
+function VerifyMobileNumber({
+  memory,
+  setMemory,
+  next,
+}: IVerifyMobileNumberProps) {
+  const { t } = useTranslation();
+  const [otp, setOtp] = useState("");
+
+  const handleSubmit = async () => {
+    setMemory((prev) => ({ ...prev, isLoading: true }));
+
+    const { res, data, error } = await request(
+      routes.abdm.healthId.abhaCreateVerifyMobileNumber,
+      {
+        body: {
+          transaction_id: memory?.transactionId,
+          otp: otp,
+        },
+      },
+    );
+
+    if (res?.status === 200 && data) {
+      setMemory((prev) => ({
+        ...prev,
+        transactionId: data.transaction_id,
+      }));
+      Notify.Success({
+        msg: data.detail ?? t("mobile_otp_verify_success"),
+      });
+      next();
+    } else {
+      Notify.Error({
+        msg: data?.detail ?? error?.message ?? t("mobile_otp_verify_error"),
+      });
+    }
+
+    setMemory((prev) => ({ ...prev, isLoading: false }));
+  };
+
+  return (
+    <div>
+      <div className="mt-0">
+        <label
+          htmlFor="mobile_number"
+          className="block text-base font-normal text-secondary-900"
+        >
+          {t("enter_mobile_number")}
+        </label>
+        <PhoneNumberFormField
+          labelSuffix={<></>}
+          name="mobile_number"
+          value={memory?.mobileNumber}
+          disabled={true}
+          onChange={() => null}
+          types={["mobile"]}
+        />
+      </div>
+
+      <div className="mt-4">
+        <OtpFormField
+          name="otp"
+          onChange={(value) => setOtp(value as string)}
+          value={otp}
+          label={t("enter_mobile_otp")}
+          disabled={memory?.isLoading}
+        />
+      </div>
+
+      <div className="mt-4 flex items-center">
+        <ButtonV2
+          className="w-full"
+          loading={memory?.isLoading}
+          onClick={handleSubmit}
+        >
+          {t("verify_otp")}
+        </ButtonV2>
+      </div>
+    </div>
+  );
+}
+
+type IChooseAbhaAddressProps = InjectedStepProps<Memory> & {
+  onSuccess: (abhaNumber: AbhaNumberModel) => void;
+};
+
+function ChooseAbhaAddress({
+  memory,
+  setMemory,
+  onSuccess,
+}: IChooseAbhaAddressProps) {
+  const { t } = useTranslation();
+  const [healthId, setHealthId] = useState("");
+  const [suggestions, setSuggestions] = useState<string[]>([]);
+
+  useEffect(() => {
+    const fetchSuggestions = async () => {
+      const { res, data } = await request(
+        routes.abdm.healthId.abhaCreateAbhaAddressSuggestion,
+        {
+          body: {
+            transaction_id: memory?.transactionId,
+          },
+        },
+      );
+
+      if (res?.status === 200 && data) {
+        setMemory((prev) => ({ ...prev, transactionId: data.transaction_id }));
+        setSuggestions(data.abha_addresses);
+      }
+    };
+
+    fetchSuggestions();
+  }, [healthId, memory?.transactionId, setMemory]);
```

Comment:
We could use `useQuery` with `onResponse` to handle the after effects.
Using `useQuery` would always ensure duplicate concurrent requests are aborted.

Reply:
Actually, `routes.abdm.healthId.abhaCreateAbhaAddressSuggestion` is a POST request